mycore-mods/src/main/java/org/mycore/mods/MCRMODSCommands.java

```diff
@@ -178,9 +182,17 @@ private static MCRDerivate createDerivate(MCRObjectID documentID, File fileDir)
         linkId.setReference(documentID, null, null);
         derivate.getDerivate().setLinkMeta(linkId);
 
+        final Predicate<Path> isDirectory = Files::isDirectory;
+        final Path rootPath = fileDir.toPath();
+        final Optional<String> firstRegularFile = Files.list(rootPath).filter(isDirectory.negate())
```

Comment:
why not use [`Files::isRegularFile`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#isRegularFile-java.nio.file.Path-java.nio.file.LinkOption...-l)?